cmd/zc_traverser_file.go

```diff
@@ -145,7 +149,7 @@ func (t *fileTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 			WarnStdoutAndScanningLog(fmt.Sprintf(invalidNameErrorMsg, targetURLParts.DirectoryOrFilePath))
 			return common.EAzError.InvalidBlobOrWindowsName()
 		}
-		if t.trailingDot != common.ETrailingDotOption.Enable() && strings.HasSuffix(targetURLParts.DirectoryOrFilePath, ".") {
+		if !t.trailingDot.IsEnabled() && strings.HasSuffix(targetURLParts.DirectoryOrFilePath, ".") {
```

Comment:
Unrelated thought, but something we should consider.  This is definitely a @gapra-msft type question as well... Should we be trimming the trailing dot if we're letting cx shoot themselves in the foot?

It's possible they actually want to properly fetch the data anyway, but that's always possible as `azcopy copy "sourcefile." "destination"`

Reply:
@adreed-msft if you're talking about the `isEnabled()` method, it lets users to use `Enable` and `AllowToUnsafeDestination`

Reply:
Discussed offline; this is in reference to functionality. Currently, if we copy a directory containing a file `file.` with the `AllowToUnsafeDestination` option enabled, `file.` will fail to transfer as it is not a valid filename on the destination.

If a customer was trying to copy one of these files, which they would be, knowingly enabling AllowToUnsafeDestination, they would likely want `file.` to become `file` when travelling to the unsafe destination.

Reply:
oh I see, so you mean "should we strip the trailing dot for customers on AllowToUnsafeDEstinations for unsafe destinations".

Reply:
I think for this particular bug, what has been implemented is sufficient - we just allow the destination filesystem to do what it will with the trailing dot